testhelpers/serialization_helper.go

```diff
@@ -36,8 +36,20 @@ func isYAMLSubsetInner(assert *assert.Assertions, expected, actual interface{},
 			}
 			expectedValueValue := expectedValue.MapIndex(keyValue)
 			actualValueValue := actualValue.MapIndex(convertedKeyValue)
-			thisPrefix := append(prefix, keyValue.String())
-			if assert.True(actualValueValue.IsValid(), "missing key %s in YAML path %s", keyValue.String(), yamlPath) {
+			// keyValue.String() does *not* return the contained value, but fmt has magic for reflect.Value types
+			thisPrefix := append(prefix, fmt.Sprintf("%s", keyValue))
+			if !actualValueValue.IsValid() {
+				if expectedValueValue.Elem().Type().Kind() == reflect.Bool && expectedValueValue.Elem().Bool() == false {
+					continue
+				}
+				if expectedValueValue.Elem().Type().Kind() == reflect.Int && expectedValueValue.Elem().Int() == 0 {
+					continue
+				}
+				if expectedValueValue.Elem().Type().Kind() == reflect.String && expectedValueValue.Elem().String() == "" {
+					continue
+				}
```

Comment:
Still not happy about skipping all those things.

Reply:
This code has been removed by https://github.com/SUSE/fissile/commit/74259420ddd948ffefb43ef3fc0abda0cba2351d